src/target/riscv/riscv.c

```diff
@@ -941,7 +948,31 @@ static int maybe_add_trigger_t2_t6(struct target *target,
 					"against the first address of the range.");
 		info->range_trigger_fallback_encountered = true;
 	}
-	return ERROR_OK;
+
+	return ret;
+}
+
+static int maybe_add_trigger_t2_t6_for_bp(struct target *target,
+		struct trigger *trigger, struct match_triggers_tdata1_fields fields)
+{
+	LOG_TARGET_DEBUG(target, "trying to setup equality match trigger");
+	struct trigger_request_info eq = {
+		.tdata1 = fields.common | fields.size.any | fields.chain.disable |
+			fields.match.eq,
+		.tdata2 = trigger->address,
+		.tdata1_ignore_mask = fields.tdata1_ignore_mask
+	};
+
+	return try_setup_single_match_trigger(target, trigger, eq);
+}
+
+static int maybe_add_trigger_t2_t6(struct target *target,
+		struct trigger *trigger, struct match_triggers_tdata1_fields fields)
+{
+	if (trigger->is_execute)
+		return maybe_add_trigger_t2_t6_for_bp(target, trigger, fields);
+
+	return maybe_add_trigger_t2_t6_for_wp(target, trigger, fields);
```

Comment:
If you like:

To make the code easier to read, you can add an `else` branch to clearly show the reader that both of the calls are logically on the same level of importance.

Assertions can be added to catch programmer mistakes and help readers to understand the code.

```suggestion
	if (trigger->is_execute) {
		assert(!trigger->is_load && !trigger->is_store);
		return maybe_add_trigger_t2_t6_for_bp(target, trigger, fields);
	} else {
		assert(trigger->is_load || trigger->is_store);
		return maybe_add_trigger_t2_t6_for_wp(target, trigger, fields);	
	}
```

Reply:
I can't add this else statements because of `tools/checkpatch.sh`:

```
WARNING:UNNECESSARY_ELSE: else is not generally useful after a break or return
#192: FILE: src/target/riscv/riscv.c:975:
+		return maybe_add_trigger_t2_t6_for_bp(target, trigger, fields);
+	} else {

```

Reply:
`assert`'s added.

Reply:
> I can't add this else statements because ...

Ah, OK, understood. This is then one of the cases where checkpatch forces somewhat worse code style, IMO.